docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1348 @@
+Looking at our little Math library we can see that both `Add` methods throw an exception if a certain condition is met not so obvious though
+is that both `Divide` methods will throw as well if the parameter `b` is zero. Let's go ahead to add exception documentation these methods.
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// <para>The main <c>Math</c> class.</para>
+/// <para>Contains all methods for performing basic math functions.</para>
+/// </summary>
+public class Math
+{
+    /// <summary>
+    /// Adds two integers and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two integers.
+    /// </returns>
+    /// <example>
+    /// <code>
+    /// int c = Math.Add(4, 5);
+    /// if (c > 10)
+    /// {
+    ///     Console.WriteLine(c);
+    /// }
+    /// </code>
+    /// </example>
+    /// <exception cref="System.OverflowException">Thrown when one parameter is max and the other is greater than 0.</exception>
+    public static int Add(int a, int b)
+    {
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+
+    /// <summary>
+    /// Adds two doubles and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two doubles.
+    /// </returns>
+    /// <exception cref="System.OverflowException">Thrown when one parameter is max and the other is greater than zero.</exception>
+    public static double Add(double a, double b)
+    {
+        if ((a == double.MaxValue && b > 0) || (b == double.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+
+    /// <summary>
+    /// Divides an integer by another and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The division of two integers.
+    /// </returns>
+    /// <exception cref="System.DivideByZeroException">Thrown when a division by zero occurs.</exception>
+    public static int Divide(int a, int b)
+    {
+        return a / b;
+    }
+
+    /// <summary>
+    /// Divides a double by another and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The division of two doubles.
+    /// </returns>
+    /// <exception cref="System.DivideByZeroException">Thrown when a division by zero occurs.</exception>
+    public static double Divide(double a, double b)
+    {
+        return a / b;
+    }
+}
+```
+
+The `cref` attribute represents a reference to an exception that is available from the current compilation environment.
+This can be any type defined in the project or a referenced assembly, the compiler will issue a warning if its value cannot be resolved.
+
+### &lt;see&gt;
+
+While documenting your code with XML tags you might reach a point where you need to add some sort of reference to another part of the code to make your reader understand it better.
```

Comment:
Here you use 2nd person. It draws the reader in to the article, and it reads like a conversation.